### ISSUE TITLE
Module::CoreList mentions Pod::Functions::Functions, but no such module has ever existed

### ISSUE THREAD

**Issue by choroba** [Perl/perl5#19991]
Module: Module::Corelist

**Description**
Module::CoreList mentions Pod::Functions::Functions, but there's no such module and there never has been. See https://www.perlmonks.org/?node_id=11145693 for the initial report.

**Steps to Reproduce**
Run:
corelist Pod::Functions::Functions
Output:
Data for 2022-08-20
Pod::Functions::Functions was first released with perl v5.20.0

**Expected behavior**
Output:
Data for 2022-08-20
Pod::Functions::Functions was not in CORE (or so I think)

**Perl configuration**

```
Summary of my perl5 (revision 5 version 37 subversion 3) configuration:
  Derived from: b5df4e081edac26a6cbaaba4f88a107f863737ca
  Platform:
    osname=linux
    osvers=5.3.18-150300.59.81-default
    archname=x86_64-linux-thread-multi
    uname='linux silent 5.3.18-150300.59.81-default #1 smp sat jul 9 12:59:06 utc 2022 (a2f05f3) x86_64 x86_64 x86_64 gnulinux '
    config_args='-Dusedevel -des -Dusethreads -Dprefix=/home/choroba/localperl'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2'
    optimize='-O2'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='7.5.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/x86_64-suse-linux/lib /usr/lib /lib64 /usr/lib64 /lib /usr/local/lib64
    libs=-lpthread -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.31.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.31'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    PERL_USE_DEVEL
    PERL_USE_SAFE_PUTENV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Locally applied patches:
    uncommitted-changes
  Built under linux
  Compiled at Jul 25 2022 00:57:15
  @INC:
    /home/choroba/localperl/lib/site_perl/5.37.3/x86_64-linux-thread-multi
    /home/choroba/localperl/lib/site_perl/5.37.3
    /home/choroba/localperl/lib/5.37.3/x86_64-linux-thread-multi
    /home/choroba/localperl/lib/5.37.3
    /home/choroba/localperl/lib/site_perl

```

**Comment by hvds** [comment]
This first appeared with f50587e06e for v5.15.8; I guess the reason it says "was first released with perl v5.20.0" is because there was a previous attempt to remove it with bd67a7ab13, but it came straight back with the next Module-CoreList update in 7520a923f1.

Have you checked whether running the tool to update CoreList will bring it back yet again after #19992? If it would, I think that should be addressed as part of the same PR - see `Porting/release_managers_guide.pod` to see how people are instructed to run it.

(Interestingly, bd67a7ab13 also removed `XSLoader::XSLoader` as another "ghost" entry, and that one did not come back; so it's possible that the resurrection of P::F::F was a manual error and that #19992 really is all that's needed.)

**Comment by choroba** [comment]
@hvds You are right, it's added by the update script. I debugged it and the name is created from `ext/Pod-Functions/Functions_pm.PL`.

I experimented with the update script and found a simple solution. I rebased my change onto the fix and force pushed into this PR, please review again.

**Comment by hvds** [comment]
> You are right, it's added by the update script. I debugged it and the name is created from `ext/Pod-Functions/Functions_pm.PL`.

Goodo.

> I experimented with the update script and found a simple solution. I rebased my change onto the fix and force pushed into this PR, please review again.

It looks like an elegant solution. I hope someone else can review it, as I've never used either Module::CoreList or the tool that updates it - my question was based purely on digging into git history. Either the last (@atoomic) or next (@neilb) scheduled release manager might be the ideal person.

**Comment by pebolle** [comment]
#19992 was merged in commit f863a123fcf1fdff2257577fb7ab92fbabc69bd5 so I guess this issue can be closed. (If the commit description had included "Fixes #19991" that would have happened automagically, I think.)

**Comment by jkeenan** [comment]
Confirmation:
```
[perlmonger: perl-5.41.2] $ ./bin/corelist Pod::Functions::Functions

Data for 2024-07-20
Pod::Functions::Functions was first released with perl v5.20.0
[perlmonger: perl-5.41.2] $ cd ../perl-5.41.3
[perlmonger: perl-5.41.3] $ ./bin/corelist Pod::Functions::Functions

Data for 2024-08-29
Pod::Functions::Functions was not in CORE (or so I think)
```
